axon-server-connector/src/main/java/org/axonframework/axonserver/connector/command/AxonServerCommandBus.java

```diff
@@ -629,12 +666,24 @@ private void processCommand(Command command) {
                 return subscriberStreamObserver;
             }
 
-            StreamObserver<CommandProviderInbound> commandsFromRoutingServer = new StreamObserver<CommandProviderInbound>() {
+            StreamObserver<CommandProviderInbound> commandsFromRoutingServer = new UpstreamAwareStreamObserver<CommandProviderInbound>() {
                 @Override
                 public void onNext(CommandProviderInbound commandToSubscriber) {
                     logger.debug("Received command from server: {}", commandToSubscriber);
-                    if (commandToSubscriber.getRequestCase() == CommandProviderInbound.RequestCase.COMMAND) {
-                        commandExecutor.execute(new CommandProcessingTask(commandToSubscriber.getCommand()));
+                    switch (commandToSubscriber.getRequestCase()) {
```

Comment:
This implementation is different from QueryService and PlatformService.

Reply:
You are right, I'll make it consistent.